platforms/enterprise/enterprise-operations/src/main/java/org/gradle/operations/problems/ProblemUsageProgressDetails.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.operations.problems;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * A problem from the problem API reported via a progress event.
+ *
+ * @since 8.14
+ */
+public interface ProblemUsageProgressDetails {
+
+    /**
+     * The problem definition, i.e. the data that is independent of the usage.
+     *
+     * @since 8.14
+     */
+    ProblemDefinition getDefinition();
+
+    /**
+     * The severity of the problem.
+     * <p>
+     * The severity of a problem is a hint to the user about how important the problem is.
+     * ERROR will fail the build, WARNING will not.
+     *
+     * @since 8.14
+     */
+    String getSeverity();
+
+    /**
+     * Declares a short, but usage-dependent message for this problem.
+     *
+     * @return the contextual label, or null if not available.
+     * @since 8.14
+     */
+    @Nullable
+    String getContextualLabel();
+
+    /**
+     * Returns solutions and advice that contain context-sensitive data, e.g. the message contains references to variables, locations, etc.
+     *
+     * @since 8.14
+     */
+    List<String> getSolutions();
```

Comment:
what kind of things will we get here?
Is a String type enough ?
ie do we expect to have to 'massage' those a bit, to eg relativize some paths, or detect URLs etc

Reply:
Currently, I would expect to not have to massage those. And the API doesn't currently give you an option to provide more structured data. So we'd need to work with what we get here.